chain/chain/src/chain.rs

```diff
@@ -1038,6 +1041,94 @@ impl Chain {
         }
     }
 
+    /// Check if any orphan with missing chunks is ready to request chunks
+    pub fn check_orphans_with_missing_chunks<F>(
+        &mut self,
+        me: &Option<AccountId>,
+        mut orphans_missing_chunks: F,
+    ) where
+        F: Copy + FnMut((CryptoHash, Vec<ShardChunkHeader>)) -> (),
+    {
+        let mut orphan_hashes = vec![];
+        for (orphan_hash, orphan) in self.orphans.orphans.iter() {
+            let mut accepted_parent = false;
+            let mut accepted_parent_hash = CryptoHash::default();
+            let mut all_parents_known = true;
+            let mut prev_hash = orphan.block.header.prev_hash;
+            for _ in 0..NUM_ORPHAN_PARENTS_CHECK {
+                if prev_hash == CryptoHash::default() {
+                    // TODO This is very suspicious.
+                    // Technically, it's correct and also helps in testing.
+                    // In reality it show never happen except maybe first seconds
+                    // of creating Universe.
+                    break;
+                }
+                // 1. Look into store
+                match self.store.get_block(&prev_hash) {
```

Comment:
Get the header instead? Pulling a block is pretty expensive, and you don't seem to be using it.